src/main/java/org/mule/service/http/impl/service/client/HttpMessageLogger.java

```diff
@@ -31,17 +34,24 @@ public HttpMessageLogger(final LoggerType loggerType) {
 
   @Override
   public void onDataReceivedEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
   @Override
   public void onDataSentEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
-  private void logBuffer(Buffer buffer) {
+  private static Map<String, String> getLogContext(Connection connection) {
+    Map<String, String> logContext = (Map<String, String>) connection.getAttributes().getAttribute("logContext");
```

Comment:
As far as I could tell it is sending a different instance on each request, but it may reuse the connection for following requests. But at that point, it should have a new set of attributes given that it contains information from the requests such as the HttpTransactionContext.